tests/csrf_tests/tests.py

```diff
@@ -442,9 +442,8 @@ def _test_https_good_referer_matches_cookie_domain(self):
     def _test_https_good_referer_matches_cookie_domain_with_different_port(self):
         req = self._get_POST_request_with_token()
         req._is_secure_override = True
-        req.META['HTTP_HOST'] = 'www.example.com'
+        req.META['HTTP_HOST'] = 'www.example.com:4443'
```

Comment:
this changes the semantics of the test, though the test name makes me think it's not actually testing the intended behavior... cookies are not distinguished by port number (https://tools.ietf.org/html/rfc6265#section-8.5) so I would expect this test to have a request with a different port number than the `Referer`

Reply:
Good question, adding `:4443` might change semantics -- I'll have to double check (though the actual code in the csrf handling might be faulty in that regard). What I am rather sure though is that the removal of `SERVER_PORT` two lines below is correct either way since the `Host` header would override it in any case. So if the tests required that before, they had been wrong :)

Reply:
I think my changes to this test are correct. The function calling it looks like this:
https://github.com/django/django/blob/9fbac5064df4eb5902c48ec3760c282da6d844f4/tests/csrf_tests/tests.py#L659-L665

Ie the cookie domain is set to `.example.com` and we check if the request is allowed, even though the site is hosted on port `4443` -- effectively checking that the port doesn't matter.